### PR TITLE
Deprecate Pretoucher

### PR DESCRIPTION
**Pull request by JerryShea**
Deprecate Pretoucher #1177

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java [hunk 1]
```diff
@@ -126,27 +126,12 @@ private void assignCurrentCycle() {
...
-                    }
-                SingleChronicleQueueStore oldStore = currentCycleWireStore;
-                currentCycleWireStore = queue.storeForCycle(qCycle, queue.epoch(), earlyAcquireNextCycle || canWrite, currentCycleWireStore);
-                if (oldStore != null && oldStore != currentCycleWireStore)
-                    oldStore.close();
-            } finally {
-                if (canWrite)
-                    queue.writeLock().unlock();
-            }
+            SingleChronicleQueueStore oldStore = currentCycleWireStore;
```
Comment (alamar) [inline src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java]: After `releaseResources()`, `currentCycleWireStore` will be null, won't it?

File: src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java [hunk 2]
```diff
@@ -155,12 +140,10 @@ private void assignCurrentCycle() {
                 cycleChangedListener.accept(qCycle);
 
                 if (earlyAcquireNextCycle)
```
Comment (alamar) [inline src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java:141]: Why we are only writing a message if this flag is set? I guess this line has to go and that isDebugEnabled the one that got to stay.

**Comment by alamar** [comment]
> RollCycleMultiThreadStressPretouchEATest
> RollCycleMultiThreadStressPretouchTest

Still fail every time when checkEA() is removed - appenders/tailers are apparently not ready for EA

File: src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java]: good point

**Comment by JerryShea** [comment]
> > RollCycleMultiThreadStressPretouchEATest
> > RollCycleMultiThreadStressPretouchTest
> 
> Still fail every time when checkEA() is removed - appenders/tailers are apparently not ready for EA

Yes - absolutely. To be expected

File: src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java]: fixed

**Review by peter-lawrey — APPROVED** [review]

**Pull request closed without merging**